### PR TITLE
Refactor display_error_message to simplify.

### PR DESCRIPTION
**Pull request by will-in-wi**
Noted that Code Climate found this method complicated, and I see why. Refactor to simplify.

### CODE REVIEW

File: lib/capistrano/application.rb [hunk 1]
```diff
@@ -63,13 +63,10 @@ def top_level_tasks
 
     def display_error_message(ex)
       unless options.backtrace
-        if (loc = Rake.application.find_rakefile_location)
-          whitelist = (@imported.dup << loc[0]).map{|f| File.absolute_path(f, loc[1])}
-          pattern = %r@^(?!#{whitelist.map{|p| Regexp.quote(p)}.join('|')})@
-          Rake.application.options.suppress_backtrace_pattern = pattern
-        end
-        trace "(Backtrace restricted to imported tasks)"
+        Rake.application.options.suppress_backtrace_pattern = backtrace_pattern
```
Comment (mattbrictson) [inline lib/capistrano/application.rb]: This has the effect of setting `Rake.application.options.suppress_backtrace_pattern` to `nil` if `loc` is falsey, which wasn't the case before the refactor.

Comment (will-in-wi) [inline lib/capistrano/application.rb]: Good catch, fixed.

**Comment by mattbrictson** [comment]
Thanks, this seems like an overall improvement.

I'm not familiar with this part of the Capistrano code though, so I don't feel comfortable merging.

@leehambley can I get a second opinion on this one?

**Comment by leehambley** [comment]
Also not familiar with it, but it looks sane. If we didn't break the tests, let's merge it :)

**Pull request merged**